envknob/envknob.go

```diff
@@ -429,3 +427,22 @@ func applyKeyValueEnv(r io.Reader) error {
 	}
 	return bs.Err()
 }
+
+// GOOS reports the effective runtime.GOOS to run as.
+//
+// In practice this returns just runtime.GOOS, unless overridden by
+// test TS_DEBUG_FAKE_GOOS.
+//
+// This allows changing OS-specific stuff like the IPN server behavior
+// for tests so we can e.g. test Windows-specific behaviors on Linux.
+// This isn't universally used.
+func GOOS() string {
+	// TODO(bradfitz): add a build tag like "ts_not_in_tests" that
+	// makes this compile to just "return runtime.GOOS" in the
+	// hope that the compiler can then inline+deadcode this away
+	// and reduce binary size outside of tests. Verify that first.
+	if v := String("TS_DEBUG_FAKE_GOOS"); v != "" {
+		return v
+	}
+	return runtime.GOOS
```

Comment:
I was testing this earlier and it does appear to deadcode out; check this out:
https://godbolt.org/z/jx1h6T678

Reply:
Cool. Well, I can't check on mobile. :) I forget whether the rules are still different interpackage too. Probably not.